enterprise/backend/src/metabase_enterprise/serialization/v2/extract.clj

```diff
@@ -37,20 +42,116 @@
         chased
         (recur to-chase chased)))))
 
+(defn- escape-analysis
+  "Given a seq of collection IDs, explore the contents of these collections looking for \"leaks\". For example, a
+  Dashboard that contains Cards which are not (transitively) in the given set of collections, or a Card that depends on
+  a Card as a model, which is not in the given collections.
+
+  Returns a data structure detailing the gaps. Use [[escape-report]] to output this data in a human-friendly format.
+  Returns nil if there are no escaped values, which is useful for a test."
+  [collection-ids]
+  (let [collection-set (->> (toucan.db/select 'Collection :id [:in collection-ids])
+                            (mapcat metabase.models.collection/descendant-ids)
+                            set
+                            (set/union (set collection-ids)))
+        dashboards     (db/select 'Dashboard :collection_id [:in collection-set])
+        ;; All cards that are in this collection set.
+        cards          (reduce set/union (for [coll-id collection-set]
+                                           (db/select-ids 'Card :collection_id coll-id)))
+
+        ;; Map of {dashboard-id #{DashboardCard}} for dashcards whose cards are outside the transitive collection set.
+        escaped-dashcards  (into {}
+                                 (for [dash  dashboards
+                                       :let [dcs (db/select 'DashboardCard :dashboard_id (:id dash))
+                                             escapees (->> dcs
+                                                           (filter :card_id) ; Text cards have a nil card_id
+                                                           (filter (comp not cards :card_id))
+                                                           set)]
+                                       :when (seq escapees)]
+                                   [(:id dash) escapees]))
+        ;; {source-card-id target-card-id} the key is in the curated set, the value is not.
+        all-cards          (for [id cards]
+                             (db/select-one ['Card :id :collection_id :dataset_query] :id id))
```

Comment:
you should actually require `metabase.model.card` and use `Card` instead of `'Card` wherever you can -- quoted model symbols are there as an escape hatch to deal with circular references between model namespaces. If you're not dealing with a potential circular reference situation you should actually require the appropriate namespace. That way tools that build dependency graphs between namespaces can pick up on the dependencies this namespace has on the model namespaces

Reply:
```suggestion
                             (db/select-one [Card :id :collection_id :dataset_query] :id id))
```

Reply:
Done.